### PR TITLE
[ML_TRAIN] Update machine-learning-training.md

### PR DESCRIPTION
**Pull request by DonghakPark**
### Change Description ###

Update Machine-Learning_training.md file
- Add Identity Layer
- Add Learning Rate Scheduler & Usage
- Remove property from optimizer

### CODE REVIEW

**Review by omar-srbd — COMMENTED** [review]
Hello @DonghakPark, I have reviewed the changes you made in the file of the PR and left some suggestions for you.
Please incorporate them.
Thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 1]
```diff
@@ -411,6 +412,7 @@ Type | Key | Value | Default value | Description
 `ML_TRAIN_LAYER_TYPE_LOSS_MSE` (since 6.5)                   |                             |                             |                         | MSE loss layer
 `ML_TRAIN_LAYER_TYPE_LOSS_CROSS_ENTROPY_SIGMOID` (since 6.5) |                             |                             |                         | Cross entropy with sigmoid loss layer
 `ML_TRAIN_LAYER_TYPE_LOSS_CROSS_ENTROPY_SOFTMAX` (since 6.5) |                             |                             |                         | Cross entropy with softmax loss layer
+`ML_TRAIN_LAYER_TYPE_IDENTITY` (since 8.0)                   |                             |                             |                         | Identity Layer
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:415]: [Fix] Thank you For Review

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 2]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
-| &#xfeff;                       | decay_steps   | (float) | Decay steps                                    |
-| &#xfeff;                       | decay_rate    | (float) | Decay rate                                     |
 | `ML_TRAIN_OPTIMIZER_TYPE_ADAM` |               |         | Adam optimizer                                 |
-| &#xfeff;                       | decay_steps   | (float) | Decay steps                                    |
-| &#xfeff;                       | decay_rate    | (float) | Decay rate                                     |
 | &#xfeff;                       | beta1         | (float) | Beta1 coefficient for Adam                     |
 | &#xfeff;                       | beta2         | (float) | Beta2 coefficient for Adam                     |
 | &#xfeff;                       | epsilon       | (float) | Epsilon coefficient for Adam                   |
 
+### Learning Rate Scheduler
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:447]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 3]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+
+Learning rate scheduler determines how to adjust the learning rate. Currently Constant, Exponential and Step Type Supported.
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:449]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 4]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+
+```c
+// Create Learning Rate Scheduler
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:452]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 5]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+ml_train_lr_scheduler_h scheduler;
+ml_train_lr_scheduler_create(&scheduler, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+
+// Configure Learning Rate Scheduler
+ml_train_lr_scheduler_set_property(scheduler, "learning_rate=0.001", NULL);
+
+// Set Learning Rate Scheduler to Optimizer
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:459]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 6]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+### Learning Rate Scheduler
+
+Learning rate scheduler determines how to adjust the learning rate. Currently Constant, Exponential and Step Type Supported.
+
+```c
+// Create Learning Rate Scheduler
+ml_train_lr_scheduler_h scheduler;
+ml_train_lr_scheduler_create(&scheduler, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+
+// Configure Learning Rate Scheduler
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:456]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 7]
```diff
@@ -498,13 +531,14 @@ Epochs = 20     # Epochs
...
 Type = adam  # Optimizer : Adaptive Moment Estimation(adam)
-Learning_rate = 0.0001  # Learning rate for the optimizer
-Decay_rate = 0.96 # The decay rate for decaying the learning rate
-Decay_steps = 1000       # decay step for the exponentially decayed learning rate
 beta1 = 0.9     # beta 1 for adam
 beta2 = 0.9999  # beta 2 for adam
 epsilon = 1e-7  # epsilon for adam
 
+[LearningRateScheduler]
+type=constant
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:539]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 8]
```diff
@@ -50,6 +50,7 @@ Following are the four major components of Machine Learning Training API:
  - [Model](#model)
  - [Layer](#layer)
  - [Optimizer](#optimizer)
+ - [Learning Rate Scheduler](#learning-rate-scheduler)
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:53]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 9]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+ml_train_lr_scheduler_create(&scheduler, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+
+// Configure Learning Rate Scheduler
+ml_train_lr_scheduler_set_property(scheduler, "learning_rate=0.001", NULL);
+
+// Set Learning Rate Scheduler to Optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_set_lr_scheduler(optimizer, scheduler);
+
+// Set Optimizer to the Model
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:463]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 10]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+ml_train_model_set_optimizer(model, optimizer);
+```
+
+Following are the available properties for each learning rate scheduler type:
+
+| Type                                     | Key          | value                       | Description  |
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:469]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 11]
```diff
@@ -664,15 +698,20 @@ Creating and setting `optimizer` to a model can be done in the same manner as `l
...
-                                                    "decay_steps=1000",
-                                                    "beta1=0.002",
+status = ml_train_optimizer_set_property(optimizer, "beta1=0.002",
                                                     "beta2=0.001",
                                                     "epsilon=1e-7", NULL);
 status = ml_train_model_set_optimizer(model, optimizer);
 ```
+### Set a learning rate scheduler
+
+Creating and setting `LearningRateScheduler` to a Optimizer can be done in the same manner as `optimizer`:
```
[the remark this answers is not in the document]
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:708]: Fix

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 12]
```diff
@@ -437,15 +439,46 @@ Following are the available properties for each optimizer type:
...
+ml_train_lr_scheduler_create(&scheduler, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+
+// Configure learning rate scheduler
+ml_train_lr_scheduler_set_property(scheduler, "learning_rate=0.001", NULL);
+
+// Set learning rate scheduler to optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_set_lr_scheduler(optimizer, scheduler);
+
+// Set Optimizer to the model
```
Comment (omar-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:463]:
```suggestion
// Set optimizer to the model
```

**Review by omar-srbd — COMMENTED** [review]
Hello @DonghakPark, thank you for making the changes. However, there is another change to be done. Please check the suggestion.
Thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 12 above)
Comment (DonghakPark) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:463]: Fix~ Thank you for your Review

**Review by omar-srbd — APPROVED** [review]
@DonghakPark, thanks for your update.

@asm-foysal, I am approving this PR for merging.
